pkg/model/components/gcpcloudcontrollermanager.go

```diff
@@ -57,6 +57,21 @@ func (b *GCPCloudControllerManagerOptionsBuilder) BuildOptions(options interface
 	if ccmConfig.ClusterCIDR == "" {
 		ccmConfig.ClusterCIDR = clusterSpec.Networking.PodCIDR
 	}
+
+	if ccmConfig.Controllers == nil {
+		if clusterSpec.Networking.GCE != nil {
+			ccmConfig.Controllers = []string{
+				"*",
+				// Don't run route controller, not needed here with "native" pod IPs
+				"-route",
```

Comment:
but if the users want to run in route mode (is that possible)? you need to enable it

Reply:
> ```
> "gkenetworkparams" is not in the list of known controllers
> Error: "gkenetworkparams" is not in the list of known controllers
> ```
> 
> Ouch - so we can't exclude an unknown controller ... we should probably fix that!
> 
> Edit: Opened [kubernetes/kubernetes#116186](https://github.com/kubernetes/kubernetes/issues/116186) (and sent a fix)

It seems I've made a typo: it should be `gkenetworkparamset`

Reply:
you should only skip in native, but I think that is safer to depend on the flag `--configure-cloud-routes`

https://github.com/kubernetes/cloud-provider-gcp/blob/427c014c890a013cec192d783e00d1cb576e5662/vendor/k8s.io/cloud-provider/app/core.go#L102-L106

Reply:
Ah - thanks for letting me know re gkenetworkparamset.  I guess that supports the idea that it should be an error, because there are two possibilities for an incorrect controller name: (1) it isn't compiled into this version, or (2) typo.  If we make "not-compiled-in" more user friendly, we make "typo" worse.

I summarized what I interpret the flags should be [here](https://github.com/kubernetes/cloud-provider-gcp/pull/482#issuecomment-1450478516), and we should be testing that right now :-)